src/platform/silabs/wifi/rs911x/WifiInterface.cpp

```diff
@@ -677,68 +677,60 @@ void ProcessEvent(WifiPlatformEvent event)
         // TODO: Currently unimplemented
         break;
     case WifiPlatformEvent::kScan: {
-#ifdef SL_WFX_CONFIG_SCAN
         rsi_rsp_scan_t scan_rsp = { 0 };
-        memset(&scan_rsp, 0, sizeof(scan_rsp));
-        int32_t status = rsi_wlan_bgscan_profile(1, &scan_rsp, sizeof(scan_rsp));
+        int32_t status          = rsi_wlan_bgscan_profile(1, &scan_rsp, sizeof(scan_rsp));
 
-        if (status != RSI_SUCCESS)
-        {
-            ChipLogError(DeviceLayer, "rsi_wlan_bgscan failed: %ld ", status);
-            return;
-        }
+        VerifyOrReturn(status == RSI_SUCCESS, ChipLogError(DeviceLayer, "rsi_wlan_bgscan_profile failed: %ld", status));
+        VerifyOrReturn(wfx_rsi.scan_cb != nullptr, ChipLogError(DeviceLayer, "wfx_rsi.scan_cb is nullptr"));
 
-        if (wfx_rsi.scan_cb == NULL)
+        uint8_t nbreOfScannedNetworks = scan_rsp.scan_count[0];
+        for (int i = 0; i < nbreOfScannedNetworks; i++)
         {
-            ChipLogError(DeviceLayer, "wfx_rsi.scan_cb is NULL");
-            return;
-        }
+            rsi_scan_info_t scan      = scan_rsp.scan_info[i];
+            wfx_wifi_scan_result_t ap = { 0 };
 
-        rsi_scan_info_t * scan;
-        wfx_wifi_scan_result_t ap;
+            ap.ssid_length = strnlen(reinterpret_cast<char *>(scan.ssid), WFX_MAX_SSID_LENGTH);
 
-        for (int x = 0; x < scan_rsp.scan_count[0]; x++)
-        {
-            scan = &scan_rsp.scan_info[x];
-            // clear structure and calculate size of SSID
-            memset(&ap, 0, sizeof(ap));
-            ap.ssid_length =
-                strnlen(reinterpret_cast<char *>(scan->ssid), std::min<size_t>(sizeof(scan->ssid), WFX_MAX_SSID_LENGTH));
-            chip::Platform::CopyString(ap.ssid, ap.ssid_length, reinterpret_cast<char *>(scan->ssid));
-
-            // check if the scanned ssid is the one we are looking for
-            if (wfx_rsi.scan_ssid_length != 0 && strncmp(wfx_rsi.scan_ssid, ap.ssid, WFX_MAX_SSID_LENGTH) != 0)
+            chip::ByteSpan scannedSsid(scan.ssid, ap.ssid_length);
+            chip::MutableByteSpan outputSsid(ap.ssid, WFX_MAX_SSID_LENGTH);
+            chip::CopySpanToMutableSpan(scannedSsid, outputSsid);
+
+            // Check if the scanned ssid is the requested Ssid
+            chip::ByteSpan requestedSsid(wfx_rsi.scan_ssid, wfx_rsi.scan_ssid_length);
+            if (!requestedSsid.empty() && !requestedSsid.data_equal(scannedSsid))
```

Comment:
`!requestedSsid.data_equal(scannedSsid)`
shouldn't data_equal be true if `continue; // we found the targeted ssid.` ?

Reply:
Comment is incorrect - updating it